query-dsl/src/main/java/org/infinispan/query/dsl/QueryBuilder.java

```diff
@@ -8,10 +8,22 @@
  */
 public interface QueryBuilder<Q extends Query> extends FilterConditionBeginContext {
 
+   QueryBuilder orderBy(Expression expression, SortOrder sortOrder);
```

Comment:
I would say no. The values themselves are Comparable. The query language is supposed to be language neutral, useable in both local and remote mode. The Comparator is a java thing.